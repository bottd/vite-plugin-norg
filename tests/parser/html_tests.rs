use insta::assert_yaml_snapshot;
use rstest::rstest;
use std::fs;
use vite_plugin_norg_parser::{extract_metadata, extract_toc, transform};

#[rstest]
#[case::basic("tests/fixtures/basic.norg")]
#[case::code_blocks("tests/fixtures/code-blocks.norg")]
#[case::headings("tests/fixtures/headings.norg")]
#[case::images("tests/fixtures/images.norg")]
#[case::links("tests/fixtures/links.norg")]
fn test_norg_fixture_files(#[case] fixture_path: &str) {
    let content = fs::read_to_string(fixture_path)
        .unwrap_or_else(|_| panic!("Failed to read {fixture_path}"));
    let ast = rust_norg::parse_tree(&content)
        .unwrap_or_else(|_| panic!("Failed to parse {fixture_path}"));

    let html = transform(&ast);
    let toc = extract_toc(&ast);

    let metadata = extract_metadata(&ast);
    assert_yaml_snapshot!(fixture_path, (html, toc, metadata));
}
