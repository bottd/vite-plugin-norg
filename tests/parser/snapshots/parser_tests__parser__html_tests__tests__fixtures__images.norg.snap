---
source: tests/parser/html_tests.rs
expression: "(html, toc, metadata)"
---
- "<h1 id=\"images\">Images</h1>\n<p>Image test</p><img src=\"./relative/image.png\" alt=\"with alt text\" />"
- - level: 1
    title: Images
    id: images
- authors: Drake Bott
  created: "2025-07-10T17:51:58-0600"
  title: Images
  updated: "2025-07-10T18:00:36-0600"
  version: 1.1.1
